src/ty/path.rs

```diff
@@ -62,30 +61,30 @@ impl IntoCompact for Path {
 }
 
 impl Path {
-	/// Start building a Path with PathBuilder
-	pub fn new(ident: <MetaForm as Form>::String, module_path: <MetaForm as Form>::String) -> Result<Path, PathError> {
```

Comment:
I went back and forth on this, and decided to panic immediately with an invalid path. An `Err` will cause a panic anyway since our `fn type_info -> Type` does not return a result. So my reasoning was to panic sooner and make the builder API simpler.

Reply:
Hmm I see. One disadvantage is that testing `Path::new` becomes harder if it panics. Not impossible though.

Reply:
`from_segments` still returns a Result and most tests are for that. I've added one for `new` panicking with invalid identifiers